src/plugins/cgal/obj.cpp

```diff
@@ -0,0 +1,267 @@
+#include "obj.h"
+
+#include <array>
+#include <cassert>
+#include <fstream>
+#include <sstream>
+#include <string>
+
+#include <boost/filesystem.hpp>
+
+#include "builder.h"
+#include "obj.h"
+
+namespace possumwood {
+
+namespace {
+struct FaceIndex {
+	int v = -1;
+	int vn = -1;
+	int vt = -1;
+};
+
+struct Face {
+	std::vector<FaceIndex> indices;
+	std::size_t objectId;
+};
+
+class FaceAdaptor {
+  public:
+	class Vertices {
+	  private:
+		static int convert(const FaceIndex& fi) {
+			return fi.v - 1;
+		}
+
+	  public:
+		Vertices(const Face* face) : m_face(face) {
```

Comment:
Codacy found an issue: [Class 'Vertices' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/martin.prazak/possumwood/pullRequest?prid=6280524)